### PR TITLE
[bard] nodejs changes

### PR DESCRIPTION
**Pull request by chejennifer**
- for nodejs/query, stop using strict mode
- pass heuristic detector & sv threshold as arguments
- make relatedQuestions out of related topics

### CODE REVIEW

File: static/nodejs_server/query.ts [hunk 1]
```diff
@@ -365,6 +400,6 @@ export async function getQueryResult(
       total: getElapsedTime(startTime, endTime),
     },
   };
-
-  return { charts: processedResults, debug };
+  const relatedQuestions = getRelatedQuestions(relatedThings, place);
```
Comment (pradh) [inline static/nodejs_server/query.ts:403]: should we flag protect this?

File: static/src/server.ts [hunk 2]
```diff
@@ -306,7 +304,8 @@ app.get("/nodejs/query", (req: Request, res: Response) => {
   const apikey = (req.query.apikey as string) || "";
   const urlRoot = `${protocol}://${host}`;
   const client = (req.query.client as string) || BARD_CLIENT_URL_PARAM;
-  const mode = (req.query.mode as string) || DEFAULT_NL_MODE;
+  const mode = (req.query.mode as string) || "";
+  const svThreshold = (req.query.svThreshold as string) || "";
```
Comment (pradh) [inline static/src/server.ts:308]: varThreshold

File: static/nodejs_server/query.ts [hunk 3]
```diff
@@ -248,13 +285,10 @@ export async function getQueryResult(
...
   try {
-    nlResp = await axios
-      // Set "mode=strict" to use heuristic detector, disable using default place,
-      // use a higher SV threshold and avoid multi-verb queries
-      .post(
-        `${apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=${mode}&client=${client}`,
-        {}
-      );
+    nlResp = await axios.post(
+      `${apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=${mode}&client=${client}&detector=${QUERY_DETECTOR}&svThreshold=${svThreshold}`,
```
Comment (pradh) [inline static/nodejs_server/query.ts:289]: What does client do in NL server?

File: static/nodejs_server/query.ts (hunk 1 above)
Comment (chejennifer) [reply static/nodejs_server/query.ts:403]: Since it's returned as a separate field in the result, wondering what the reason for flag protecting it is?

Comment (pradh) [reply static/nodejs_server/query.ts:403]: This is just a prototype that we may change evolve to doing heavier-weight things, like using an LLM.  Having a flag set lets bard caller control that.  And if the flag is set, we don't need to do related-thing handling either.

File: static/nodejs_server/query.ts (hunk 3 above)
Comment (chejennifer) [reply static/nodejs_server/query.ts:289]: hm, looking through the editors, looks like client was added in this pr: https://github.com/datacommonsorg/website/pull/3785 

tracing through the code though, I'm not sure it's still being used ..

File: static/src/server.ts (hunk 2 above)
Comment (chejennifer) [reply static/src/server.ts:308]: done

File: static/nodejs_server/query.ts (hunk 1 above)
Comment (chejennifer) [reply static/nodejs_server/query.ts:403]: sure, can flag protect this for now, updated

File: static/nodejs_server/query.ts (hunk 3 above)
Comment (pradh) [reply static/nodejs_server/query.ts]: oh lol, i added it!  Ah, yes, this goes into the logs.

File: static/nodejs_server/query.ts [hunk 4]
```diff
@@ -248,13 +286,10 @@ export async function getQueryResult(
...
   try {
-    nlResp = await axios
-      // Set "mode=strict" to use heuristic detector, disable using default place,
-      // use a higher SV threshold and avoid multi-verb queries
-      .post(
-        `${apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=${mode}&client=${client}`,
-        {}
-      );
+    nlResp = await axios.post(
+      `${apiRoot}/api/explore/detect-and-fulfill?q=${query}&mode=${mode}&client=${client}&detector=${QUERY_DETECTOR}&varThreshold=${varThreshold}`,
```
Comment (pradh) [inline static/nodejs_server/query.ts:290]: TODO: do we want to control related-thing generation?

**Review by pradh — APPROVED** [review]
Thanks!

File: static/nodejs_server/query.ts (hunk 4 above)
Comment (chejennifer) [reply static/nodejs_server/query.ts]: added todo

**Pull request merged**